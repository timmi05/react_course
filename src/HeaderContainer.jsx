import React from 'react';
import { SearchPanel } from './search/SearchPanel';
import {Product } from './product/Product';
import { AppLabel } from './component/AppLabel';
import { SearchLink } from './component/SearchLink';
import './HeaderContainer.css';

export const HeaderContainer = props => {
    const { productSelected } = props;
    const Content = productSelected ? Product : SearchPanel;

    return (
        <div className='header-container'>
            <div className='header-container-label'>
                <span>
                < AppLabel />
                </span>
                <span className='header-container-search'>
                    < SearchLink
                    productSelected = { productSelected }
                    />
                </span>
            </div>
            < Content
                runtime = '145'
                releaseDate = '2019'
                description = 'Best film of year'
                title = 'Guardians of the galaxy: VOL. 3'
                tagline = 'Tagline'
                voteAverage = '4'
                posterPath = 'https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg'
                overview = "The third film based on Marvel's Guardians of the Galaxy.
                 The third film based on Marvel's Guardians of the Galaxy. The third film based on Marvel's Guardians of
                  the Galaxy. The third film based on Marvel's Guardians of the Galaxy. The third film based on Marvel's
                   Guardians of the Galaxy"
             />
         </div>
    );
};
