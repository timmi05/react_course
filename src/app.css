.app{
    background-color: #555555;
    color: #ffffff;
    margin: -8px;
}