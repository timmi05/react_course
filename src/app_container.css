.app-container{
    margin: 0 auto;
    width: 1280px;
}

.app-container-lower-panel{
    background-color: #424242;
    text-align: center;
    height: 70px;
    display: flex;
    justify-content: center;
    flex-direction: column;
}