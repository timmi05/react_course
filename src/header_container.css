.header-container{
    background: #232323;
    padding: 15px 50px 30px;

}

.header-container-label{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}
