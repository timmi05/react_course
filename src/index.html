<!DOCTYPE html>
<html>
<head>
    <meta name="viewport">
    <meta charset="UTF-8">
</head>
    <body>
        <div id="app"></div>
    </body>
</html>