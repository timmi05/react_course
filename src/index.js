import * as React from 'react';
import * as ReactDom from 'react-dom';
import {Hello} from './hello_component';

ReactDom.render(<Hello toWhat='world'/>,
document.getElementById('app'));