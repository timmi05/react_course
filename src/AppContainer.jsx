import React from 'react';
import HeaderContainer from './HeaderContainer';
import ResultContainer from './result/ResultContainer';
import AppLabel from './component/AppLabel';
import './app_container.css';

const AppContainer = () => {
     const productSelected = true;
     const genre = productSelected ? 'Drama' : null;
     return(
         <div className="app-container">
             <HeaderContainer
             productSelected = { productSelected }
             />
             <ResultContainer
             productSelected = { productSelected }
             genre = { genre }
             />
             <div className="app-container-lower-panel">
                < AppLabel />
             </div>
        </div>
    )
}

export default AppContainer