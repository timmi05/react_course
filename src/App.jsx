import React from 'react';
import { AppContainer } from './AppContainer';
import { ErrorBoundary } from  './error/ErrorBoundary'
import './App.css';

export const App = () => {
    const productSelected = true;
    const genre = productSelected ? 'Drama' : null;

    const data =
        // [];
        [{
        id: 1,
        posterPath: 'https://image.tmdb.org/t/p/w500/uJ6OnE3CzGWq6buLINAbdBqa0gV.jpg',
        title: 'Solo: A Star Wars Story',
        tagline: 'A Star Wars',
        releaseDate : '2018'
    }, {
        id:2,
        posterPath: 'https://image.tmdb.org/t/p/w500/uJ6OnE3CzGWq6buLINAbdBqa0gV.jpg',
        title: 'Solo: A Star Wars Story',
        tagline: 'A Star Wars',
        releaseDate: '2018'
    }, {
        id: 3,
        posterPath: 'https://image.tmdb.org/t/p/w500/uJ6OnE3CzGWq6buLINAbdBqa0gV.jpg',
        title: 'Solo: A Star Wars Story',
        tagline: 'A Star Wars',
        releaseDate: '2018'
    }, {
        id: 4,
        posterPath: 'https://image.tmdb.org/t/p/w500/uJ6OnE3CzGWq6buLINAbdBqa0gV.jpg',
        title: 'Solo: A Star Wars Story',
        tagline: 'A Star Wars',
        releaseDate: '2018'
    }, {
        id: 5,
        posterPath: 'https://image.tmdb.org/t/p/w500/uJ6OnE3CzGWq6buLINAbdBqa0gV.jpg',
        title: 'Solo: A Star Wars Story',
        tagline: 'A Star Wars',
        releaseDate: '2018'
    }, {
        id: 6,
        posterPath: 'https://image.tmdb.org/t/p/w500/uJ6OnE3CzGWq6buLINAbdBqa0gV.jpg',
        title: 'Solo: A Star Wars Story',
        tagline: 'A Star Wars',
        releaseDate: '2018'
    }];

    return(
        <div className="app">
            <ErrorBoundary>
                <AppContainer
                productSelected = { productSelected }
                genre = { genre }
                data = { data }
                />
            </ErrorBoundary>
        </div>
    )
};
