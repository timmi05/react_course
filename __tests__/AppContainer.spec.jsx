import React from 'react';
import { shallow } from 'enzyme'
import { AppContainer } from "../src/AppContainer";

describe('AppContainer rendering', () => {
    it('renders correctly', () => {
        const component = shallow(<AppContainer/>);
        expect(component).toMatchSnapshot();
    });

    it('renders correctly', () => {
        const component = shallow(<AppContainer
        productSelected = { true }
        genre = 'genre'
        data = { [] }
        />);
        expect(component).toMatchSnapshot();
    });
});
