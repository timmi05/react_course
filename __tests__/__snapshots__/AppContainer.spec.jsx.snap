// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer rendering renders correctly 1`] = `
<div
  className="app-container"
>
  <HeaderContainer />
  <Result />
  <div
    className="app-container-lower-panel"
  >
    <AppLabel />
  </div>
</div>
`;
